lib/parsers/aac_parser/adts_header_info.rb

```diff
@@ -0,0 +1,134 @@
+# This is a representation of the relevant information found in an Audio Data Transport Stream (ADTS) file header.
+class FormatParser::AdtsHeaderInfo
+  attr_accessor :mpeg_version, :layer, :protection_absence, :profile, :mpeg4_sampling_frequency_index,
+                :mpeg4_channel_config, :originality, :home_usage, :frame_length, :buffer_fullness,
+                :aac_frames_per_adts_frame
+
+  # An ADTS header has the following format, when represented in bits:
+  # AAAAAAAA AAAABCCD EEFFFFGH HHIJKLMM MMMMMMMM MMMOOOOO OOOOOOPP (QQQQQQQQ QQQQQQQQ)
+  # The chunks represented by these letters have specific meanings, as described here:
+  # https://wiki.multimedia.cx/index.php/ADTS
+
+  AAC_ADTS_HEADER_BITS_CHUNK_SIZES = [['A', 12], ['B', 1], ['C', 2], ['D', 1], ['E', 2], ['F', 4], ['G', 1], ['H', 3], ['I', 1], ['J', 1], ['K', 1], ['L', 1], ['M', 13], ['O', 11], ['P', 2], ['Q', 16]]
+  MPEG4_AUDIO_OBJECT_TYPE_RANGE = 0..45
+  MPEG4_AUDIO_SAMPLING_FREQUENCY_RANGE = 0..14
+  MPEG4_AUDIO_SAMPLING_FREQUENCY_HASH = {
+    0 => 96000, 1 => 88200, 2 => 64000, 3 => 48000, 4 => 44100, 5 => 32000, 6 => 24000,
+    7 => 22050, 8 => 16000, 9 => 12000, 10 => 11025, 11 => 8000, 12 => 7350,
+    13 => 'Reserved', 14 => 'Reserved'
+  }
+  AAC_PROFILE_DESCRIPTION_HASH = { 0 => 'AAC_MAIN', 1 => 'AAC_LC (Low Complexity)', 2 => 'AAC_SSR (Scaleable Sampling Rate)', 3 => 'AAC_LTP (Long Term Prediction)' }
+  MPEG_VERSION_HASH = { 0 => 'MPEG-4', 1 => 'MPEG-2'}
+
+  def mpeg4_sampling_frequency
+    if !@mpeg4_sampling_frequency_index.nil? && MPEG4_AUDIO_SAMPLING_FREQUENCY_HASH.has_key?(@mpeg4_sampling_frequency_index)
+      return MPEG4_AUDIO_SAMPLING_FREQUENCY_HASH[@mpeg4_sampling_frequency_index]
+    end
+    nil
```

Comment:
`nil` is implied to be returned, so you don't have to explicitly have it. If it's for added clarity, I'm okay with that 😊

Reply:
Let's say it's for clarity 👀